### PR TITLE
feat: assign user pr

### LINKED ISSUE #26: Assign user on PR opening
I believe that the v1 of the bot would automatically assign a user to an issue if that said user would open a pull-request against an issue. Currently this doesn't happen, for example in https://github.com/ubiquibot/command-start-stop/issues/22

I do not know how this would play with the maximum assigned issues however, if a user would be above the maximum allowed threshold.

### PR DESCRIPTION
**Pull request by gentlementlegen**
Resolves #26
QA: https://github.com/Meniole/command-start-stop/issues/14#issuecomment-2337444468

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "Start | Stop",
   "description": "Assign or un-assign yourself from an issue.",
-  "ubiquity:listeners": ["issue_comment.created", "issues.assigned"],
+  "ubiquity:listeners": ["issue_comment.created", "issues.assigned", "pull_request.opened", "pull_request.reopened"],
```
Comment (0x4007) [inline manifest.json:4]: Reopened it doesn't seem to make sense

File: src/handlers/user-start-stop.ts [hunk 2]
```diff
@@ -26,10 +28,10 @@ export async function userStartStop(context: Context): Promise<Result> {
   return { status: HttpStatusCode.NOT_MODIFIED };
 }
 
-export async function userSelfAssign(context: Context): Promise<Result> {
+export async function userSelfAssign(context: Context<"issues.assigned">): Promise<Result> {
   const { payload } = context;
   const { issue } = payload;
-  const deadline = getDeadline(issue);
+  const deadline = getDeadline(issue.labels);
```
Comment (0x4007) [inline src/handlers/user-start-stop.ts:34]: What's the deadline bit for?

**Review by 0x4007 — CHANGES_REQUESTED** [review]
We never needed reopened in the past

File: manifest.json (hunk 1 above)
Comment (gentlementlegen) [reply manifest.json:4]: I can remove it, I thought it would make sense if a PR is reopened to re-assign the user if it was not assigned anymore.

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
Just gonna focus on shipping and merging simple pulls to try and catch back up with things

**Pull request merged**